addons/sale/models/res_partner.py

```diff
@@ -43,3 +43,20 @@ def can_edit_vat(self):
             ('state', 'in', ['sent', 'sale', 'done'])
         ], limit=1)
         return can_edit_vat and not bool(has_so)
+
+    @api.depends_context('company', 'sale_order_ids')
+    def _credit_debit_get(self):
+        # Extended in order to include the amount of confirmed sales orders in the partner's credit,
+        # after withdrawing what has already been invoiced.
+        super()._credit_debit_get()
+        residual_amounts = self.env['sale.order'].read_group(
+            domain=[('state', '=', 'sale'), ('company_id', '=', self.env.company.id)],
```

Comment:
```suggestion
            domain=[('state', 'in', ('sale', 'done')), ('company_id', '=', self.env.company.id)],
```
'locked' SO are considered confirmed as well

Reply:
Regarding the inclusions of Sales Order in the computation : this is quite a hot topic even amongst POs. On one side we have the simplicity of the functionality, on the other we have the accuracy of the computation used. For what it's worth, we all agree we don't want to go anywhere else than just the display of warnings.

The task started with a very simplistic view where only the Invoiced amounts will be taken into account for the computation of the Used Credit of a partner. We gathered some feedbacks, and we have internal opinions that if the sum of confirmed SOs is not taken into account, then the view of the Partner account might not be accurate enough and we would miss some warnings. Hence the push for trying to gather those amounts as well.

My stance is that even a simplistic functionality is better than nothing. If summing the confirmed Sales Orders is deemed not worth it on the technical / computational side of things, then we can fallback on the simplest form of the functionality, even if it's not my preferred solution.

Reply:
Well, IMHO, if we start considering potentially accountable documents on the side, we have to consider all of them, not only the Sales Orders, as they are not the only ones generating Invoices (/account moves).

Also, this introduces some potentially confusing side-effects: if a SO was already paid (not invoiced, but paid through a payment acquirer), it will still be considered for the credit warning, potentially confusing the salesman :confused:.